src/main/java/org/semux/core/TransactionResult.java

```diff
@@ -181,6 +195,8 @@ public void setError(Error error) {
         for (byte[] log : logs) {
             enc.writeBytes(log);
         }
+        // todo - add fork activated check
+        enc.writeAmount(gasUsed);
```

Comment:
Please double check if the transaction hash is part of the consensus rule, secured by Merkle hash?